polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -342,7 +344,11 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				original_origin = ?self.original_origin,
 				"Trapping assets in holding register",
 			);
-			let effective_origin = self.context.origin.as_ref().unwrap_or(&self.original_origin);
+			let effective_origin = if let Some(asset_claimer) = self.asset_claimer.as_ref() {
```

Comment:
Now that the custom claimer is also in the picture, it might be better to change the name:

```suggestion
			let claimer = if let Some(asset_claimer) = self.asset_claimer.as_ref() {
```

Reply:
refactored